### PR TITLE
[feat] Drop option `--flex-alloc-tasks` and internal unused sanity function utilities

### LINKED ISSUE #1317: Revisit deprecated features and options and consider removing them completely for 3.0

### PR DESCRIPTION
**Pull request by ekouts**
Remove:
- `--flex-alloc-tasks` option from cli
- evaluate and make_deferrable in deferrable.py

The features and options that are not deprecated yet in reframe 3.0 are tracked in different issues.

Closes #1317.

### CODE REVIEW

File: docs/manpage.rst [hunk 1]
```diff
@@ -366,12 +366,6 @@ If no node can be selected, the test will be marked as a failure with an appropr
      This is the default policy.
    - Any positive integer: Flexible tests will be assigned as many tasks as needed in order to span over the specified number of nodes from the node pool.
 
-.. option:: --flex-alloc-tasks[=POLICY]
-
-   .. deprecated:: 2.21
-
-      Please use |--flex-alloc-nodes|_ instead.
-
 .. |--flex-alloc-nodes| replace:: :attr:`--flex-alloc-nodes`
```
Comment (vkarak) [inline docs/manpage.rst:369]: Can you check if this placeholder is used elsewhere in the page and remove it if not?

Comment (ekouts) [reply docs/manpage.rst]: It's not used anywhere else in this page, so I removed it.

File: docs/manpage.rst [hunk 2]
```diff
@@ -366,13 +366,6 @@ If no node can be selected, the test will be marked as a failure with an appropr
...
    - Any positive integer: Flexible tests will be assigned as many tasks as needed in order to span over the specified number of nodes from the node pool.
 
-.. option:: --flex-alloc-tasks[=POLICY]
-
-   .. deprecated:: 2.21
-
-      Please use |--flex-alloc-nodes|_ instead.
-
-.. |--flex-alloc-nodes| replace:: :attr:`--flex-alloc-nodes`
 .. _--flex-alloc-nodes: #cmdoption-flex-alloc-nodes
```
Comment (vkarak) [inline docs/manpage.rst:369]: You need to remove this line, too. Sorry, the syntax of Sphinx is really silly if you want to format a link...

Comment (ekouts) [reply docs/manpage.rst]: No problem, removed this as well

**Pull request merged**